lib/kafo/data_types/boolean.rb

```diff
@@ -0,0 +1,27 @@
+module Kafo
+  module DataTypes
+    class Boolean < DataType
+      def typecast(value)
+        case value
+          when '0', 'false', 'f', 'n', false
+            false
+          when '1', 'true', 't', 'y', true
+            true
+          else
+            value
+        end
+      end
+
+      def valid?(input, errors = [])
+        if input.is_a?(::TrueClass) || input.is_a?(FalseClass)
```

Comment:
I think this code would be more readable as
```ruby
valid = input.is_a?(::TrueClass) || input.is_a?(FalseClass)
errors << "#{input.inspect} is not a valid boolean" unless valid
valid
```

seems a bit more readable, sadly there's no #tap in 1.8.7 which would make it one-liner

also, why `::TrueClass` but `FalseClass` without absolute namespace path